src/main/scala/it/unibo/authsim/library/user/model/CryptoInformation.scala

```diff
@@ -0,0 +1,21 @@
+package it.unibo.authsim.library.user.model
+
+import it.unibo.authsim.library.dsl.policy.model.Policy
+
+/**
+ * Trait that represent a CryptoIntormation
+ */
+trait CryptoInformation(algorithmPolicy:Policy)
+/*
+NB Abbiamo deciso che viene criptata solo la password ->
+per me qui ci sta di più l'algoritmo(hash, (a)symmetric estenderanno da un trait comune 'CryptographicAlgorithm') che la policy
+ */
+
+/**
+ * Object that represent a CryptoIntormation
+ */
+object CryptoInformation:
+  //TODO documentation when implementation is stable
+  def apply(algorithmPolicy: Policy): CryptoInformation = CryptoInformationImpl(algorithmPolicy)
+  //non sono sicura abbia senso tenere una case class per sta roba
+  case class CryptoInformationImpl(algorithmPolicy: Policy) extends CryptoInformation(algorithmPolicy)
```

Comment:
This case class can be made private as users should call `apply` of the companion object.

About the usefulness of the `case class`, for me is better to have a case class rather than an anonymous extension

Reply:
Oks